### ISSUE TITLE
Move non-blockchain Ethereum common tests out of blockchain app

### ISSUE THREAD

**Issue by germsvel** [poanetwork/mana#293]
Currently we have the common StateTests and TransactionTests inside the `blockchain` app:
* https://github.com/poanetwork/mana/blob/master/apps/blockchain/test/blockchain/state_test.exs and 
* https://github.com/poanetwork/mana/blob/master/apps/blockchain/test/blockchain/transaction_test.exs. 

We should probably move those out of the blockchain app since they aren't the blockchain tests.  

We also have ethereum common `BlockchainTests` and `VMTests`. Those reside within their respective apps: 
* https://github.com/poanetwork/mana/blob/master/apps/blockchain/test/blockchain_test.exs. 
* https://github.com/poanetwork/mana/blob/master/apps/evm/test/evm_test.exs

I'm not sure where we should move the common tests. Clearly, not all of them match our apps 1-1 (e.g. we do not have an `apps/transaction`. Because of this, it may make sense to have all the ethereum common tests in a single place (even the blockchain and evm ones). If that does not sound like a good idea, we should at the very least move the StateTests and TransactionTests outside of the blockchain app.

**Comment by hayesgm** [comment]
#547 Begins to at least address unifying our common test code framework. Are you suggesting that `Transaction.ex` and its counterparts belong in a different umbrella-app all together?

**Comment by hayesgm** [comment]
Relates to #566

**Comment by germsvel** [comment]
@hayesgm 

> Are you suggesting that Transaction.ex and its counterparts belong in a different umbrella-app altogether

No, I hope we _don't_ move the implementations out to separate umbrella apps. 

I think this issue was a little confused and didn't have a clear suggestion. It stemmed from the fact that our ethereum common tests are mixed in with our regular tests, and that can seem strange sometimes. 

So maybe a better aim of this issue would be to figure out whether or not we want to have all ethereum common tests live in a single place (just the tests not the implementations) or whether we prefer just keeping them as they are. I think it could provide some clarity on what are ethereum common tests and what are `mana`'s own tests. 

All this being said, I think after having more knowledge of the code base and if #566 actually happens, I think they could stay where they are. In which case, we could close this issue altogether.
